### PR TITLE
Add limited connection type support in ForesightManager and UI settings

### LINKED ISSUE #15: Allow configurable isLimitedConnection threshold (e.g., 2G → 3G)
**Description:**
Currently, `isLimitedConnection` is hardcoded to detect a 2G network. We’d like the ability to configure this threshold so it can be set to 3G or higher.

**Use Case:**
In our webshop, we want certain functionality to only run when a 3G+ connection is detected, ensuring a better user experience for customers with faster internet speeds.

**Proposed Solution:**

* Add a configuration option to define the minimum network type for `isLimitedConnection`.
* Allow values such as `2g`, `3g`, `4g`, etc.

**Example:**

```js
ForesightManager.initialize({
  minNetworkType: '3g'
```

### PR DESCRIPTION
**Pull request by stijns96**
Add the possibility to change the minimum required internet connection for element to be registered.

This is a bit against how the package worked, because that was based on a limitation while I think that it would make more sense to test it on a minimum connection speed and not a fixed value.

I also experienced this with another developer that was confused by the fact that it was based on a 2G connection while in theory it should already work on 3G.

Closes #15

### CODE REVIEW

File: packages/docs/versioned_docs/version-3.3/configuration/global-settings.md [hunk 1]
```diff
@@ -204,6 +205,23 @@ ForesightManager.initialize({
...
 
+### Minimum Connection Settings (v3.3.0+)
+
+#### `minimumConnectionType`
+
+- **Type:** `MinimumConnectionType`
+- **Default:** `"3g"`
+- **Options:** `"slow-2g"`, `"2g"`, `"3g"`, `"4g"`
+- **Description:** The minimum connection speed required to register elements. Elements will not be registered when the user's connection is slower than this threshold.
+
```
Comment (spaansba) [inline packages/docs/versioned_docs/version-3.3/configuration/global-settings.md:216]: lets add a note that it wont work for all users since it is not baseline yet

File: packages/docs/docs/configuration/element-settings.md [hunk 2]
```diff
@@ -195,7 +195,7 @@ const { isTouchDevice, isLimitedConnection, isRegistered } = ForesightManager.in
 #### `isLimitedConnection`
 
 - **Type:** `boolean`
-- **Description:** Is `true` when the user is on a 2G connection or has data-saver enabled. Elements will not be registered when connection is limited.
+- **Description:** Is `true` when the user's connection matches the configured `minimumConnectionType` setting (defaults to "3g"). Elements will not be registered when connection is limited. See [Global Settings](/docs/configuration/global-settings#minimumconnectiontype) for details.
```
Comment (spaansba) [inline packages/docs/docs/configuration/element-settings.md:198]: Lets mention it wont work for all users yet

File: packages/js.foresight/src/helpers/shouldRegister.ts [hunk 3]
```diff
@@ -39,5 +42,15 @@ function hasConnectionLimitations(): boolean {
...
+
+  // Define array of connection types from slowest to fastest
+  const connectionTypes: MinimumConnectionType[] = ["slow-2g", "2g", "3g", "4g"]
+  // Get index of user's current connection speed in the array (e.g. "4g" would be index 3)
+  const currentConnectionIndex = connectionTypes.indexOf(connection.effectiveType as MinimumConnectionType)
+  // Get index of the minimum connection speed required in settings (e.g. "3g" would be index 2)
+  const minimumConnectionIndex = connectionTypes.indexOf(minimumConnectionType)
+
+  // If user's connection is slower than the minimum required, or data saver is enabled, return true
+  return currentConnectionIndex <= minimumConnectionIndex || connection.saveData
```
Comment (spaansba) [inline packages/js.foresight/src/helpers/shouldRegister.ts:55]: Since its minimum I dont think it should return true if they are equal
